SQL_2709_String_Functions.sql

```diff
@@ -0,0 +1,113 @@
+----------------------------------STRING FUNCTIONS------------------------------------------------
+
+--1. Extract the middle 3 characters from the string 'ABCDEFG'.
+SELECT SUBSTRING('ABCDEFG',3,3);
+
+--2. From a table 'Employees' with a column 'FullName', write a query to extract the first name (assuming it's always the first word before a space).
+SELECT LEFT (FullName,CHARINDEX(' ',FullName)-1) AS FirstName FROM Employees;
+--SELECT STRING_SPLIT (FullName , ' ') AS FirstName FROM Employees;;
+--3. Extract the first 5 characters from the string 'SQL Server 2022'.
+SELECT LEFT ('SQL Server 2022',5);
+
+--4. From a 'Products' table with a 'ProductCode' column, write a query to get the first 3 characters of each product code
+SELECT LEFT (ProductCode,5) from Products;
+
+--5. Extract the last 4 characters from the string 'ABCDEFGHIJKL'.
+SELECT RIGHT ('ABCDEFGHIJKL',4);
+
+--6. From an 'Orders' table with an 'OrderID' column (format: ORD-YYYY-NNNN), write a query to extract just the numeric portion at the end.
+SELECT RIGHT (OrderID,4) FROM Orders;

```

Comment:
How can we write this query generically, for example : if the ORDER ID is ORD-2023-123467 or ORD-2023-1234741.

Reply:
SELECT RIGHT(OrderID, CHARINDEX('-',REVERSE(OrderID));
would work for a numeric portion of any size

Reply:
Will this work ?

Reply:
should work if i add a '-1' and fix the parentheses :

SELECT RIGHT(OrderID,CHARINDEX('-',REVERSE(OrderID))-1) ;